### ISSUE TITLE
microsoft.data.sqlclient is not supported on this platform

### ISSUE THREAD

**Issue by chomelchizo** [dotnet/SqlClient#634]
Local Azure Functions no longer working after update to 2.0. As soon as SqlConnection is instantiated within a function, the exception pops up.

macOS Catalina v 10.15.5
Visual Studio for Mac Community v 8.6.5

Framework

- Microsoft.NETCore.App(3.1.0) dotnet --version returns 3.1.301

Packages

- Microsoft.Azure.WebJobs.Extension.Storage (4.02)
- Microsoft.SqlData.SqlClient (2.0.0)
- Microsoft.Sdk.Functions(3.0.8)

**Once deployed to Azure, functions operate flawlessly.**

**Comment by JRahnama** [comment]
@chomelchizo Thanks for bringing this up to us. I have couple of questions/request to understand the situation better:

1. Are you using Azure CLI on your local machine?
2. From my understanding your application was working fine with Local Azure Functions using any versions prior to 2.0.0. Is that 
    correct?
3. I can see you are using the latest version of Microsoft.Sdk,Functions (3.0.8) which has been released 4 days ago. Can you 
   kindly try your application with versions prior to that release as well?
4. The very last question, Is it possible to provide us some sample application to work with?

Thanks.

**Comment by chomelchizo** [comment]
> 1. Are you using Azure CLI on your local machine?
**_Yes. (https://docs.microsoft.com/en-us/azure/azure-functions/functions-develop-vs#testing-functions) I test and debug the function app right from VS for Mac, which runs within a terminal. 
Azure Functions Core Tools 3.0.2358 
Function Runtime Version: 3.0.13159.0_**

> 2. From my understanding your application was working fine with Local Azure Functions using any versions prior to 2.0.0. Is that  correct?
**_Yes. All functions work fine once deployed to Azure._**

> 3. I can see you are using the latest version of Microsoft.Sdk,Functions (3.0.8) which has been released 4 days ago. Can you kindly try your application with versions prior to that release as well?
**_I could, but not for a few days. Sorry._**

> 4. The very last question, Is it possible to provide us some sample application to work with?
**_Same response as 3._**

**Comment by chomelchizo** [comment]
@JRahnama 

Downgraded Microsoft.Sdk.Functions from 3.0.8 to 3.0.7 and local execution of functions worked.

**Comment by veatcha** [comment]
Can also confirm this same error with local execution is addressed by Microsoft.NET.Sdk.Functions downgrade from 3.0.8 to 3.0.7.

Windows 10
Azure Functions Core Tools: 3.0.2630
Function Runtime:  3.0.13901.0
Target Framework: netcoreapp3.1

**Comment by mwwhited** [comment]
Changed target platform to AnyCPU and added <RuntimeIdentifier>win10-x64</RuntimeIdentifier> to the PropertyGroup in my csproj... 3.0.8 + SqlClient now works for me in the azure function emulator.

**Comment by cheenamalhotra** [comment]
Closing this issue as it comes from external component.

**Comment by matthewblott** [comment]
I'm not using any other library. I'm just trying to create a simple connection in a Xamarin Workbook.

```
var conn = new Microsoft.Data.SqlClient.SqlConnection(connectionString)
```

The above code throws the same 'unsupported platform' error.

Framework: .NET Core 3.1
OS: macOS Catalina
Model Name:	MacBook Pro
Processor Name:	Quad-Core Intel Core i7
Processor Speed:	2.2 GHz
Number of Processors:	1
Total Number of Cores:	4
Memory:	16 GB

**Comment by cheenamalhotra** [comment]
![image](https://user-images.githubusercontent.com/13396919/88418575-9ebb0080-cd98-11ea-8a1f-233f69aef98c.png)

**Comment by matthewblott** [comment]
![s 2020-07-24 at 18 45 18](https://user-images.githubusercontent.com/1231417/88419923-e87a0a00-cddd-11ea-8615-c57e0e6fc52e.jpg)

**Comment by cheenamalhotra** [comment]
Have you tested with Microsoft.Data.SqlClient v2.0.0 as well ?

**Comment by matthewblott** [comment]
Yes that's the version I was using. Strange, the same code works with Rider though!

![s 2020-07-24 at 18 53 39](https://user-images.githubusercontent.com/1231417/88420618-0f850b80-cddf-11ea-95f8-f43ab07c704b.jpg)

**Comment by cheenamalhotra** [comment]
I also tested older versions till **1.0.19239.1** and all work for me.
Works in Visual Studio for Mac, Visual Studio Code (dotnet cli).

What were you running on when you see PlatformNotSupported?

**Comment by matthewblott** [comment]
```Environment.OSVersion.Platform``` returns ```Unix``` even though it's macOS. But that's the same if I'm using Rider or Xamarin Workbooks. It seems to be an issue with Xamarin Workbooks by the looks of it. I haven't had any issues until now.

**Comment by matthewblott** [comment]
> I also tested older versions till **1.0.19239.1** and all work for me.
> Works in Visual Studio for Mac, Visual Studio Code (dotnet cli).
> 
> Maybe an issue with Xamarin workbooks?

Yeah, it seems like it. Thanks for getting back so promptly, help appreciated :-)

**Comment by matthewblott** [comment]
Okay, I thought I'd add this as it might be relevant if someone else ends up here. If you set the Console type to System Mono instead of .NET Core in Xamarin Workbooks everything works as expected. I have no idea why this is, though I noticed when i run the workbooks terminal binary it hits some path problems which seem to be due to the fact I don't have the full Xamarin installed (I installed Workbooks only). Anyway, I hope this helps someone :-)

![s 2020-07-24 at 20 05 21](https://user-images.githubusercontent.com/1231417/88426401-06993780-cde9-11ea-85e4-6f0ed6727497.jpg)
